internal/asm/arm64/impl.go

```diff
@@ -2899,283 +2899,260 @@ func (a *AssemblerImpl) EncodeStaticConstToVectorRegister(n *NodeImpl) (err erro
 // advancedSIMDAcrossLanes holds information to encode instructions as "Advanced SIMD two-register miscellaneous" in
 // https://developer.arm.com/documentation/ddi0596/2021-12/Index-by-Encoding/Data-Processing----Scalar-Floating-Point-and-Advanced-SIMD?lang=en
 var advancedSIMDTwoRegisterMisc = map[asm.Instruction]struct {
-	U, Opcode byte
-	// TODO: extract common implementation of qAndSizeResolver.
-	qAndSizeResolver func(arrangement VectorArrangement) (Q, Size byte, err error)
```

Comment:
I just did the refactoring again, but nothing interesting... just decided to use map instead of anonymous functions
as this was getting hairy :D